pkg/dockerregistry/auth/openshift.go

```diff
@@ -0,0 +1,180 @@
+package auth
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	log "github.com/Sirupsen/logrus"
+	ctxu "github.com/docker/distribution/context"
+	registryauth "github.com/docker/distribution/registry/auth"
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/dockerregistry"
+	"golang.org/x/net/context"
+)
+
+func init() {
+	registryauth.Register("openshift", registryauth.InitFunc(newAccessController))
+}
+
+type contextKey int
+
+var bearerTokenKey contextKey = 0
+
+func WithBearerToken(parent context.Context, bearerToken string) context.Context {
+	return context.WithValue(parent, bearerTokenKey, bearerToken)
+}
+
+func BearerTokenFrom(ctx context.Context) (string, bool) {
+	bearerToken, ok := ctx.Value(bearerTokenKey).(string)
+	return bearerToken, ok
+}
+
+type AccessController struct {
+}
+
+var _ registryauth.AccessController = &AccessController{}
+
+type authChallenge struct {
+	err error
+}
+
+var _ registryauth.Challenge = &authChallenge{}
+
+type OpenShiftAccess struct {
```

Comment:
I'm not sure I like this being a struct. Could we remove it and modify `VerifyOpenShiftAccess` to take in the 4 fields as arguments instead?